internal/kafka/internal/config/kafka_supported_instance_types.go

```diff
@@ -0,0 +1,202 @@
+		return fmt.Errorf("maxDataRetentionSize for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	maxDataRetentionPeriod, err := duration.ParseISO8601(k.MaxDataRetentionPeriod)
+	if err != nil {
+		return fmt.Errorf("maxDataRetentionPeriod for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	if maxDataRetentionPeriod.IsZero() || egressThroughputQuantity.CmpInt64(1) < 0 ||
+		ingressThroughputQuantity.CmpInt64(1) < 0 || maxDataRetentionSize.CmpInt64(1) < 0 ||
+		k.TotalMaxConnections <= 0 || k.MaxPartitions <= 0 || k.MaxConnectionAttemptsPerSec <= 0 ||
+		k.QuotaConsumed < 1 || k.CapacityConsumed < 1 {
+		return fmt.Errorf("Kafka instance size '%s' for instance type '%s' specifies a property value less than or equals to Zero.", k.Id, instanceTypeId)
+	}
+
+	return nil
+}
+
+type SupportedKafkaInstanceTypesConfig struct {
+	SupportedKafkaInstanceTypes []KafkaInstanceType `yaml:"supported_instance_types"`
+}
+
+func (s *SupportedKafkaInstanceTypesConfig) validate() error {
+	existingInstanceTypes := make(map[string]int, len(s.SupportedKafkaInstanceTypes))
+
+	for _, KafkaInstanceType := range s.SupportedKafkaInstanceTypes {
+		if _, ok := existingInstanceTypes[KafkaInstanceType.Id]; ok {
+			return fmt.Errorf("Kafka instance type id '%s' was defined more than once.", KafkaInstanceType.Id)
+		}
+		existingInstanceTypes[KafkaInstanceType.Id]++
+
+		if err := KafkaInstanceType.validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type KafkaSupportedInstanceTypesConfig struct {
+	Configuration     SupportedKafkaInstanceTypesConfig
+	ConfigurationFile string
+}
+
+func NewKafkaSupportedInstanceTypesConfig() *KafkaSupportedInstanceTypesConfig {
+	return &KafkaSupportedInstanceTypesConfig{
+		ConfigurationFile: "config/kafka-instance-types-configuration.yaml",
+	}
+}
+
+func (s *SupportedKafkaInstanceTypesConfig) GetKafkaInstanceTypeByID(instanceType string) (*KafkaInstanceType, error) {
+	for _, t := range s.SupportedKafkaInstanceTypes {
+		if t.Id == instanceType {
+			ret := t
+			return &ret, nil
+		}
+	}
+	return nil, fmt.Errorf("Unable to find kafka instance type for '%s'", instanceType)
+}
+
+func (kp *KafkaInstanceType) GetKafkaInstanceSizeByID(sizeId string) (*KafkaInstanceSize, error) {
+	for _, size := range kp.Sizes {
+		if size.Id == sizeId {
+			ret := size
+			return &ret, nil
+		}
+	}
+	return nil, fmt.Errorf("Kafka instance size id: '%s' not found for '%s' instance type", sizeId, kp.Id)
+}
+
+type Plan string
+
+func (p Plan) String() string {
+	return string(p)
+}
+
+func (p Plan) GetInstanceType() (string, error) {
+	t := strings.Split(strings.TrimSpace(p.String()), ".")
+	if len(t) != 2 {
+		return "", errors.New(errors.ErrorGeneral, fmt.Sprintf("Unsupported plan provided: '%s'", p))
+	}
+	return t[0], nil
+}
+
+func (p Plan) GetSizeID() (string, error) {
+	t := strings.Split(strings.TrimSpace(p.String()), ".")
+	if len(t) != 2 {
```

Comment:
have number of plans as a constant, to avoid future maintenance work when plans increase/decrease

Reply:
see above, this refers to the format of the plan.

Reply:
See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/965#discussion_r859506885 comment